fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java

```diff
@@ -257,7 +260,22 @@ public void delete() {
             final Iterator<Property> inboundProperties = Iterators.concat(references, weakReferences);
 
             while (inboundProperties.hasNext()) {
-                inboundProperties.next().remove();
+                final Property prop = inboundProperties.next();
+                final List<Value> newVals = new ArrayList<Value>();
+                final Iterator<Value> propIt = JcrPropertyFunctions.property2values.apply(prop);
+                while (propIt.hasNext()) {
```

Comment:
Oh, I didn't know that `property2values` did that-- that's lame. It's not worth worrying about this any more anyway- this all goes away with Java 8's Streams, so this is limited-lifetime code. I'm sorry I took up this much time with it.